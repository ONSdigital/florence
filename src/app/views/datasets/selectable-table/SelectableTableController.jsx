import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import dateFormat from 'dateformat';

import url from '../../../utilities/url';

const propTypes = {
    values: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string.isRequired,
        date: PropTypes.string,
        id: PropTypes.string.isRequired,
        instances: PropTypes.arrayOf(PropTypes.shape({
            date: PropTypes.string,
            edition: PropTypes.string.isRequired,
            version: PropTypes.oneOfType([
                PropTypes.string.isRequired,
                PropTypes.number.isRequired
            ]),
            status: PropTypes.string
        }))
    }))
}

export default class SelectableTableController extends Component {
    constructor(props) {
        super(props);
    }
    renderDatasetInstances(instances) {
        return instances.map((instance, i) => {
            if (instance !== undefined) {

                return (
                    <div key={i} className="grid simple-table__row">
                        <div className="grid__col-2">
                            {/* Wrap in span to stop grid col from forcing link to be full width 
                            (which made focus outline much wider than the link itself) */}
                            <span><Link to={instance.url}>View</Link></span>
                        </div>
                        <div className="grid__col-3">
                            {instance.status}
                        </div>
                        <div className="grid__col-3">
                            {dateFormat(instance.date, "dd/mm/yy HH:MM")}
                        </div>
                        <div className="grid__col-2">
                            {instance.edition}
                        </div>
                        <div className="grid__col-1">
                            {instance.version}
                        </div>
                    </div>
                )
            }
        })
    }

    render() {
        return (
            <div className="selectable-table">
                    <div className="selectable-table__heading">
                        Title
                    </div>
                    {this.props.values.map((item, index) => {
                        return (
                            <details key={index} className="selectable-table__details">
                                <summary className="selectable-table__summary">
                                    <div className="grid">
                                        <div className="grid__col-10">
                                            <strong>{item.title}</strong>
                                        </div>
                                    </div>
                                </summary>
                                <div>
                                    <Link className="inline-block margin-bottom--1" to={url.resolve(`datasets/${item.id}/metadata`)}> Edit dataset details</Link>
                                    {item.instances.length > 0 &&
                                        <div>
                                            <div className="grid simple-table__heading">
                                                <div className="grid__col-2">
                                                    
                                                </div>
                                                <div className="grid__col-3">
                                                    Status
                                                </div>
                                                <div className="grid__col-3">
                                                    Date last modified
                                                </div>
                                                <div className="grid__col-2">
                                                    Edition
                                                </div>
                                                <div className="grid__col-1">
                                                    Version
                                                </div>
                                            </div>
                                            { this.renderDatasetInstances(item.instances) }
                                        </div>
                                    }
                                </div>
                            </details>
                        )
                    })}
            </div>
        )
    }
}

SelectableTableController.propTypes = propTypes;
