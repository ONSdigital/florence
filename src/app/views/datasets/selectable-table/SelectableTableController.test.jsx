import React from 'react';
import SelectableTableController from './SelectableTableController.jsx';
import renderer from 'react-test-renderer';

const values = [
    {
        title: "CPI",
        date: "17-10-2017",
        datasetURL: "/florence/datasets/12345/metadata",
        instances: [
            {
                date: "",
                edition: "2016",
                version: "-",
                url: "/florence/datasets/12345/instances/6789/metdata"
            }
        ]
    }
]

test('Selectable table with values renders component with all values', () => {
    const component = renderer.create(
        <SelectableTableController 
            values={values}
        />
    );

    expect(component.toJSON()).toMatchSnapshot();
});