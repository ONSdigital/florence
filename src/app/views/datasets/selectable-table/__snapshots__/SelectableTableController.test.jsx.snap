// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectable table with values renders component with all values 1`] = `
<div
  className="selectable-table"
>
  <div
    className="selectable-table__heading"
  >
    <ul
      className="selectable-table__list"
    >
      <li
        className="selectable-table__list-item grid__col-6"
      >
        Title
      </li>
      <li
        className="selectable-table__list-item grid__col-6"
      >
        Submission Date
      </li>
    </ul>
  </div>
  <div>
    <details
      className=""
      onClick={[Function]}
      open={false}
    >
      <summary>
        <ul
          className="selectable-table__list"
        >
          <li
            className="selectable-table__list-item grid__col-6"
          >
            <strong>
              CPI
            </strong>
          </li>
          <li
            className="selectable-table__list-item grid__col-4"
          >
            17-10-2017
          </li>
          <li
            className="selectable-table__list-item grid__col-2"
          >
            <div
              className="selectable-table__accordion-closed"
            />
          </li>
        </ul>
      </summary>
      <div
        className="selectable-table__list"
      >
        <a
          href="/florence/datasets/12345/metadata"
        >
           Edit dataset details 
        </a>
        <ul>
          <li
            className="selectable-table__list-item grid__col-4"
          >
            Date
          </li>
          <li
            className="selectable-table__list-item grid__col-2"
          >
            Edition
          </li>
          <li
            className="selectable-table__list-item grid__col-2"
          >
            Version
          </li>
          <hr />
        </ul>
        <div>
          <ul>
            <li
              className="selectable-table__list-item grid__col-4"
            >
              
            </li>
            <li
              className="selectable-table__list-item grid__col-2"
            >
              2016
            </li>
            <li
              className="selectable-table__list-item grid__col-4"
            >
              -
            </li>
            <li
              className="selectable-table__list-item grid__col-2"
            >
              <a
                href="/florence/datasets/12345/instances/6789/metdata"
              >
                View
              </a>
            </li>
            <hr />
          </ul>
        </div>
      </div>
    </details>
    
  </div>
</div>
`;
